tensorflow/contrib/tensorrt/convert/convert_nodes.cc

```diff
@@ -939,6 +941,21 @@ Status Converter::RenameAndMarkOutputTensors(
     if (tensor == nullptr) {
       return errors::NotFound("Output tensor not found: ", output.first);
     }
+    // Check if this tensor has already been marked as an output.
+    // ConvertIdentity can cause the same tensor to be repeated in
+    // output_tensors, which can cause us to overwrite the name of the output
+    // tensor binding. For example, if we rename OutputPH_0 to OutputPH_1 then
+    // we won't be able to locate OutputPH_0 during runtime. To fix this,
+    // duplicate the tensor using no-op shuffle.
+    // TODO(tmorris): Remove this work-around once we use TRT's IIdentityLayer
+    // in ConvertIdentity.
+    if (tensorflow::str_util::StartsWith(tensor->getName(), kOutputPHName)) {
+      nvinfer1::IShuffleLayer* layer = network()->addShuffle(*tensor);
```

Comment:
Add a comment or expand TODO to mention that we are using shuffle layer for identity by not setting any transpose or reshape.